environments/openshift/main.jsonnet

```diff
@@ -0,0 +1,52 @@
+local list = import 'telemeter/lib/list.libsonnet';
+
+// This entire file takes what we have for Kubernetes and
+// generates an OpenShift specific Template on top of that.
+
+local app =
+  (import 'kube-thanos.libsonnet') +
+  {
+    _config+:: {
+      namespace: '{$NAMESPACE}',
+
+      images+: {
+        thanos: '$IMAGE',
+      },
+    },
+
+    // This generates the Template kind that OpenShift requires
+
+    local t = super.thanos,
+    thanos+:: {
+      template:
+        local objects = {
+          ['querier-' + name]: t.querier[name]
+          for name in std.objectFields(t.querier)
+        } + {
+          ['store-' + name]: t.store[name]
+          for name in std.objectFields(t.store)
+        };
+
+        list.asList('thanos', objects, [
+          {
+            name: 'NAMESPACE',
+            value: 'telemeter',
```

Comment:
How do these values interact with the config values? Should we have:
```
value: $._config.namespace,
```
as the default value for the variable as we do in the telemeter repo?

Reply:
Yes. But for simplicity reasons I've set `$._config.namespace` to `${NAMESPACE}`. Especially overwriting the `dnssrv+` flags becomes quite tricky otherwise.
In the end, for all kubernetes/ksonnet things, we should probably not even use `$._config`.